src/Main.cpp

```diff
@@ -0,0 +1,299 @@
+                   configMINIMAL_STACK_SIZE + 3000, NULL, tskIDLE_PRIORITY + 1,
+                   NULL);
+  if(s1 != pdPASS) 
+    if(firmware_mode == fw_debug) Serial.printf("S1 creation problem\r\n");
+  s2 = xTaskCreate(imu_task, "imu_task",
+                   configMINIMAL_STACK_SIZE + 1000, NULL, tskIDLE_PRIORITY + 1,
+                   NULL);
+  if(s2 != pdPASS) 
+    if(firmware_mode == fw_debug) Serial.printf("S2 creation problem\r\n");
+  s3 = xTaskCreate(runtime_stats_task, "runtime_stats_task",
+                   configMINIMAL_STACK_SIZE + 1000, NULL, tskIDLE_PRIORITY + 1,
+                   NULL);
+  if(s3 != pdPASS) 
+    if(firmware_mode == fw_debug) Serial.printf("S3 creation problem\r\n");
+  s4 = xTaskCreate(pid_handler_task, "pid_handler_task",
+                            configMINIMAL_STACK_SIZE + 1000, NULL, tskIDLE_PRIORITY + 3,
+                            NULL);
+  if(s4 != pdPASS)  
+    if(firmware_mode == fw_debug) Serial.printf("S4 creation problem\r\n");
+  s5 = xTaskCreate(pixel_led_task, "pixel_led_task",
+                          configMINIMAL_STACK_SIZE + 1000, NULL, tskIDLE_PRIORITY + 1,
+                          NULL);
+  if(s5 != pdPASS)  
+    if(firmware_mode == fw_debug) Serial.printf("S5 creation problem\r\n");
+  s7 = xTaskCreate(BoardSupportTask, "BoardSupportTask",
+                          configMINIMAL_STACK_SIZE + 500, NULL, tskIDLE_PRIORITY + 1,
+                          NULL);
+  if(s7 != pdPASS) 
+    if(firmware_mode == fw_debug) Serial.printf("S7 creation problem\r\n");
+  s8 = xTaskCreate(power_board_task, "power_board_task",
+                          configMINIMAL_STACK_SIZE + 500, NULL, tskIDLE_PRIORITY + 1,
+                          NULL);
+  if(s8 != pdPASS) 
+    if(firmware_mode == fw_debug) Serial.printf("S8 creation problem\r\n"); 
+  s9 = xTaskCreate(uros_ping_agent_task, "uros_ping_agent_task",
+                        configMINIMAL_STACK_SIZE + 100, NULL, tskIDLE_PRIORITY + 1,
+                        NULL);
+  if(s9 != pdPASS) 
+    if(firmware_mode == fw_debug) Serial.printf("S9 creation problem\r\n");
+  /* HARDWARE ACTIONS BEFORE RTOS STARTING */
+  SetGreenLed(On);
+  SetRedLed(Off);
+  /* START RTOS */
+  if(firmware_mode == fw_debug) Serial.printf("Tasks starting\r\n");
+  vTaskStartScheduler();
+}
+
+static void RclcSpinTask(void *p) {
+  UNUSED(p);
+  TickType_t xLastWakeTime = xTaskGetTickCount();
+  static uRosFunctionStatus uRosPingAgentStatus;
+  while(1) {
+    xQueueReceive(uRosPingAgentStatusQueue, &uRosPingAgentStatus, (TickType_t) 0); 
+    vTaskDelayUntil(&xLastWakeTime, 1);
+    switch (uRosLoopHandler(uRosPingAgentStatus))
+    {
+    case Ok:
+      SetGreenLed(On);
+      SetRedLed(Off);
+      break;
+    case Error:
+      SetGreenLed(Off);
+      SetRedLed(On);
+      vTaskDelay(100);
+      SetRedLed(Off);
+      vTaskDelay(100);
+      break;
+    default:
+      break;
+    }
+  }
+}
+
+static void imu_task(void *p){
```

Comment:
```suggestion
static void ImuTask(void *p){
```
And similar in other functions